pkg/trace/sampler/config_watcher.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2015 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package sampler
+
+import (
+	"fmt"
+	"github.com/DataDog/datadog-agent/pkg/trace/metrics"
+	"time"
+)
+
+type ConfigWatcher struct {
+	prioritySampler *PrioritySampler
+	errorsSampler   *ErrorsSampler
+	rareSampler     *RareSampler
+	stop            chan struct{}
+}
+
+func NewConfigWatcher(prioritySampler *PrioritySampler, errorsSampler *ErrorsSampler, rareSampler *RareSampler) *ConfigWatcher {
+	return &ConfigWatcher{
+		prioritySampler: prioritySampler,
+		errorsSampler:   errorsSampler,
+		rareSampler:     rareSampler,
+		stop:            make(chan struct{}),
+	}
+}
+
+func (w *ConfigWatcher) Start() {
```

Comment:
Please pay extra attention here since this is my first time doing this kind of things in Go 🙏